main/main.cpp

```diff
@@ -29,38 +33,59 @@ esp_err_t set_adc(
 	adc_channel_t  adc_channel
 );
 
-QueueHandle_t refer_speed_q = xQueueCreate(2, sizeof(float));
-QueueHandle_t motor_speed_q = xQueueCreate(2, sizeof(float));
-QueueHandle_t motor_count_q = xQueueCreate(2, sizeof(int));
+QueueHandle_t refer_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t motor_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t motor_count_q = xQueueCreate(1, sizeof(int));
+
+QueueHandle_t tel_ref_speed_q   = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_motor_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_error_q       = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_error_der_q   = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_control_signal_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_exec_time_q      = xQueueCreate(1, sizeof(float));
 
 typedef struct {
 	adc_oneshot_unit_handle_t adc_handle;
-	// volatile uint32_t *motor_counter;
 } timer_args;
 
 void IRAM_ATTR send_status(void* argp) {
```

Comment:
Pretty complex code for an ISR. ISR are desing to only notify events not doing calculations. all calculations MUST be done outside in other task. ISR only need to handle events and send by queues or eventGroupWaitBits.